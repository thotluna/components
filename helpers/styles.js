import classNames from 'classnames'

export const getDynamicClasses = (cssModule = {}, props = {}, classes) => {
  return classes.reduce((classesObject, classKey) => {
    const propValue = props[classKey]
    const className = cssModule[`${classKey}-${propValue}`]

    return className && propValue
      ? { ...classesObject, [className]: propValue }
      : classesObject
  }, {})
}

export const getModuleClasses = (cssModule, classKey) => {
  return (cssModule && cssModule[classKey]) || classKey
}

export const getObjectClasses = (cssModule, object) => {
  return Object.keys(object).reduce((classes, classKey) => {
    const className = cssModule[classKey]
    return className ? { ...classes, [className]: object[classKey] } : classes
  }, {})
}

export const getClasses = (cssModule = {}) => (props) => (...args) => {
  const allClass = args.map((arg) => {
    if (Array.isArray(arg)) {
      const dinamicClasses = getDynamicClasses(cssModule, props, arg)
      // console.log('>>>dinamic', dinamicClasses)
      return dinamicClasses
    } else if (typeof arg === 'string') {
      const moduleClass = getModuleClasses(cssModule, arg)
      // console.log('>>>module', moduleClass)
      return moduleClass
    } else if (typeof arg === 'object') {
      const objectClass = getObjectClasses(cssModule, arg)
      // console.log('>>>object', objectClass)
      return objectClass
    }
  })
  // console.log(allClass)

  const result = classNames(allClass)

  // console.log(result)
  return result
}
