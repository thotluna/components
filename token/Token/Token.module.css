.container {
  display: inline-flex;
  flex-direction: row;
  align-items: center;
}

.description {
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  margin-left: 10px;
  gap: 4px;
}

.label {
  min-width: max-content;
  padding: 4px 8px;
  border: 1px solid var(--button-base-color-border);
  background-color: var(--color-base-transparent);
  border-radius: var(--border-radius-sm);
  color: var(--color-font-base);
  font-family: var(--font-family-mono);
  font-size: var(--font-size-sm);
}

.label:first-child {
  border: var(--border-width-thin) solid var(--button-base-color-border);
  background-color: var(--color-base-transparent);
}

.color {
  width: 65px;
  height: 65px;
  border: var(--button-base-color-border);
  border-radius: var(--border-radius-full);
  box-shadow: var(--box-shadow-sm);
}

.spacer {
  display: flex;
  width: 50px;
  height: 50px;
  align-items: center;
  justify-content: center;
}

.font-size,
.font-family {
  width: 100%;
  max-width: 500px;
  text-align: justify;
}

.font-size {
  text-align: right;
}

.title {
  font-family: var(--font-family-title);
  font-size: 16px;
}

.sans {
  font-family: var(--font-family-sans);
  font-size: 16px;
}

.box {
  width: 20rem;
  height: 5rem;
  border: 1px solid black;
}
