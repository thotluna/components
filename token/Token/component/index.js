export { default as Color } from './Colors/Colors'
export { default as Spacing } from './Spacing/Spacing'
export { default as FontFamily } from './Fonts/FontFamily'
export { default as FontSize } from './Fonts/FontSize'
export { default as FontWeight } from './Fonts/FontWeight'
