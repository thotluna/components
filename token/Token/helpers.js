import toKebabCase from '../../utils/toKebalCase'

export const getTokenItems = (tokens, tokenKey) => (group) => {
  if (group) {
    return Object.keys(tokens[group]).map((key) => ({
      label: `--${toKebabCase(tokenKey)}-${group}-${toKebabCase(key)}`,
      value: tokens[group][key],
    }))
  }

  return Object.keys(tokens).map((key) => ({
    label: `--${toKebabCase(tokenKey)}-${toKebabCase(key)}`,
    value: tokens[key],
  }))
}
