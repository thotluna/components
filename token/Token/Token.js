import React from 'react'

import { Color, Spacing, FontFamily, FontSize, FontWeight } from './component'

const Token = () => <></>

Token.Color = Color
Token.Spacing = Spacing
Token.FontFamily = FontFamily
Token.FontSize = FontSize
Token.FontWeight = FontWeight

export default Token
