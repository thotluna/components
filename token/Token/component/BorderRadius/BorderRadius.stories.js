import Token from '../..'
import { choices } from '../../..'

import { getTemplate, getListTemplate } from '../../../../helpers/storybook'
import { getTokenItems } from '../../helpers'

export default {
  title: 'Tokens/BorderRadius',
  component: Token.BorderRadius,
  args: {
    children:
      "Depend noticed bones Edge disturbing flesh heard cream. Morninged facial judgment Hobbit fortress. Camp flatten buy dagger Sit. Leagues sweeter 1400 mean friend unlost DÃ©agol trade stuffing? Smithy's jiffy merely treacherous cozy interesting ask seat Thror's.",
  },
  argTypes: {
    value: {
      control: {
        type: 'inline-radio',
        options: Object.keys(choices.borderRadius).map(
          (key) => choices.borderRadius[key]
        ),
      },
    },
    isPlayground: {
      table: {
        disable: true,
      },
    },
    label: {
      table: {
        disable: true,
      },
    },
    children: {
      table: {
        disable: true,
      },
    },
  },
}

const Template = getTemplate(Token.BorderRadius)

export const Default = Template.bind({})
Default.args = {
  label: '--border-radius-none',
  value: '0',
}

const getItems = getTokenItems(choices.borderRadius, 'borderRadius')
const ListTemplate = getListTemplate(Token.BorderRadius)

export const Borders = ListTemplate.bind({})
Borders.args = {
  items: getItems(),
}
