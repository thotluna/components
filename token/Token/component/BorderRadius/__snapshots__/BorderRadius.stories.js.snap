// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Tokens/BorderRadius Borders 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "background": "",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
      "padding": "2em",
    }
  }
>
  <div
    className="container"
  >
    <div
      className="box"
      style={
        Object {
          "borderRadius": "0",
        }
      }
    />
    <div
      className="description"
    >
      <span
        className="label"
      >
        --border-radius-none
      </span>
      <span
        className="label"
      >
        0
      </span>
    </div>
  </div>
  <div
    className="container"
  >
    <div
      className="box"
      style={
        Object {
          "borderRadius": "10px",
        }
      }
    />
    <div
      className="description"
    >
      <span
        className="label"
      >
        --border-radius-sm
      </span>
      <span
        className="label"
      >
        10px
      </span>
    </div>
  </div>
  <div
    className="container"
  >
    <div
      className="box"
      style={
        Object {
          "borderRadius": "20px",
        }
      }
    />
    <div
      className="description"
    >
      <span
        className="label"
      >
        --border-radius-md
      </span>
      <span
        className="label"
      >
        20px
      </span>
    </div>
  </div>
  <div
    className="container"
  >
    <div
      className="box"
      style={
        Object {
          "borderRadius": "30px",
        }
      }
    />
    <div
      className="description"
    >
      <span
        className="label"
      >
        --border-radius-lg
      </span>
      <span
        className="label"
      >
        30px
      </span>
    </div>
  </div>
  <div
    className="container"
  >
    <div
      className="box"
      style={
        Object {
          "borderRadius": "9999px",
        }
      }
    />
    <div
      className="description"
    >
      <span
        className="label"
      >
        --border-radius-full
      </span>
      <span
        className="label"
      >
        9999px
      </span>
    </div>
  </div>
</div>
`;

exports[`Storyshots Tokens/BorderRadius Default 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "background": "",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
      "padding": "2em",
    }
  }
>
  <div
    className="container"
  >
    <div
      className="box"
      style={
        Object {
          "borderRadius": "0",
        }
      }
    />
    <div
      className="description"
    >
      <span
        className="label"
      >
        --border-radius-none
      </span>
      <span
        className="label"
      >
        0
      </span>
    </div>
  </div>
</div>
`;
